service/history/shard/context_impl.go

```diff
@@ -80,16 +103,22 @@ type (
 
 		// exist only in memory
 		remoteClusterCurrentTime map[string]time.Time
-
-		// true if previous owner was different from the acquirer's identity.
-		previousShardOwnerWasDifferent bool
 	}
 )
 
 var _ Context = (*ContextImpl)(nil)
 
-// ErrShardClosed is returned when shard is closed and a req cannot be processed
-var ErrShardClosed = errors.New("shard closed")
+var (
+	// ErrShardClosed is returned when shard is closed and a req cannot be processed
+	ErrShardClosed = errors.New("shard closed")
+
+	// ErrShardStatusUnknown means we're not sure if we have the shard lock or not. This may be returned
+	// during short windows at initialization and if we've lost the connection to the database.
+	ErrShardStatusUnknown = errors.New("shard status unknown")
```

Comment:
One more question: this (eventually) gets returned from grpc handlers, so I think it has to be a serviceerror instead? Should ErrShardClosed also be a serviceerror?

Reply:
this should be ShardOwnershipLost or Unavailable